pkg/bid/parser.go

```diff
@@ -0,0 +1,245 @@
+package bid
+
+import (
+	"fmt"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+)
+
+func ParseBid(bidStr string) (BID, error) {
+	rs := &BidScanner{str: bidStr}
+
+	tType, val, err := rs.NextToken()
+	if err != nil {
+		return nil, err
+	}
+	if tType != literal || val != BidPrefix {
+		return nil, fmt.Errorf("invalid baton id prefix: %s", val)
```

Comment:
I think we should make an error to return from this package:

The easiest thing to do is like
```
var ErrInvalidBID = errors.New("bid: invalid")
```

Then everywhere wrap it:
```
return nil, fmt.Errorf("%w: missing baton id prefix", ErrInvalidBID)
```

Then callers can actually test for it with `errors.Is` and whatnot.

We could alternatively make a custom error that could include the index of the parse error and stuff, but v0v.